homeassistant/components/mobile_app/entity.py

```diff
@@ -0,0 +1,135 @@
+"""A entity class for mobile_app."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.const import (CONF_WEBHOOK_ID, STATE_ON, STATE_OFF)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import (ATTR_SENSOR_ATTRIBUTES,
+                    ATTR_SENSOR_DEVICE_CLASS, ATTR_SENSOR_ICON,
+                    ATTR_SENSOR_NAME, ATTR_SENSOR_STATE, ATTR_SENSOR_TYPE,
+                    ATTR_SENSOR_TYPE_BINARY_SENSOR, ATTR_SENSOR_UNIQUE_ID,
+                    ATTR_SENSOR_UOM, DATA_REGISTRATIONS, DOMAIN,
+                    SIGNAL_SENSOR_UPDATE)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_mobile_app_platform(sensor_type, hass, config,
+                                          async_add_entities,
+                                          discovery_info=None):
+    """Set up the mobile app entity sensor."""
+    registrations = hass.data[DOMAIN][DATA_REGISTRATIONS]
+    sensor_configs = hass.data[DOMAIN][sensor_type]
+
+    platform_name = "MobileAppEntity"
+
+    if sensor_type == ATTR_SENSOR_TYPE_BINARY_SENSOR:
+        platform_name = "MobileAppBinarySensor"
+
+    platform = globals()[platform_name]
+
+    entities = list()
+
+    if discovery_info is None:
```

Comment:
When will this happen?

Reply:
In webhook.py, after a sensor is registered for the first time, we call `load_platform`.